MagicBots/Overseer/Framework/OverseerModule.cs

```diff
@@ -0,0 +1,29 @@
+﻿using Discord.WebSocket;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+
+namespace MagicBots.Overseer.Framework
+{
+    public abstract class OverseerModule
+    {
+        protected readonly IConfiguration Config;
+        protected readonly DiscordSocketClient Discord;
+        protected readonly ILogger Logger;
+
+        protected OverseerModule(DiscordSocketClient discord, ILogger logger,
```

Comment:
How does DiscordSocketClient get injected in here?

Reply:
If you look at the next [`simple-modules`](https://github.com/monodop/magic-bots/pull/7) PR it gets injected via `OverseerHostedService.Client` right now, but should be refactored to use something better in the future. Added a todo to improve this mechanism.